src/coordination/coordinator_cluster_state.cpp

```diff
@@ -208,10 +210,39 @@ auto CoordinatorClusterState::GetInstanceUUID(std::string_view instance_name) co
   return it->second.instance_uuid;
 }
 
-auto CoordinatorClusterState::IsLockOpened() const -> bool {
+auto CoordinatorClusterState::GetIsLockOpened() const -> bool {
   auto lock = std::shared_lock{log_lock_};
   return is_lock_opened_;
 }
 
+void CoordinatorClusterState::SetIsLockOpened(bool is_lock_opened) {
+  auto lock = std::unique_lock{log_lock_};
+  is_lock_opened_ = is_lock_opened;
+}
+
+void CoordinatorClusterState::SetReplicationInstances(
+    std::map<std::string, ReplicationInstanceState, std::less<>> const &replication_instances_) {
+  auto lock = std::unique_lock{log_lock_};
+  repl_instances_ = replication_instances_;
+}
+
+void CoordinatorClusterState::SetCurrentMainUUID(utils::UUID const &current_main_uuid) {
+  auto lock = std::unique_lock{log_lock_};
+  current_main_uuid_ = current_main_uuid;
+}
```

Comment:
I think you can accept by value and std::move into member variables. Also can you writ just a small comment why we need these 3 methods. E.g methods used for nlohmann

Reply:
Changed. I have used it this way because copy constructors accept params by const &, so it looked as a nice rule to follow. I added a general comment as we can stop using that method for nlohmann function